generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -213,6 +223,7 @@ int main(int argc, char** argv)
             verilog_plusargs_legal = 0;
```

Comment:
Actually the code here will not distinguish whether the removed arguments are outside the permissive section or not. It will simply copy all arguments defined in the plug-args list generated from Verilog or the HTIF argument lists, as well as the arguments `+permissive` and `+permissive-off`. The only two consumers of arguments passed into the Verilator environment now are HTIF parser and Verilog plus-args, and we will need to modify `emulator.cc` anyway if we add another parser to make sure that they are not considered invalid arguments.

Reply:
Never mind, I will need to look at this field more closely.

Reply:
I moved `verilated_argv` to the 'P' clause of the `switch` code block. If we have a plusargs, it will only leaves the last two if-else clauses by setting `c` to either 'P' (when `emulator.cc` does not detect any issue) or '?' (when `emulator.cc` will report an illegal argument and terminate the simulation). This ensure that all arguments not recognized in `emulator.cc` will be added to `verilated_argv` unless it is an illegal argument, in which case the simulation terminated directly. 
The remaining plusargs defined in `emulator.cc` are not used anywhere else except for `+permissive` and `+permissive-off` which is used by HTIF parser, therefore I must put those two into `verilated_argv`.

Reply:
I just found that we can still leave those two if-else clauses by `goto done_processing`, and I am looking at this right now.

Reply:
Fixed. Hitting HTIF arguments will make the program exit the while loop, and I have code after the while loop that copies all the remaining arguments (which can only be HTIF) and the binary name onto the verilator argument stack.